views/components/ConfigurationCard/index.jsx

```diff
@@ -0,0 +1,187 @@
+
+    if (status === 200) loadAllConfigurations(uid)
+  }
+
+  const removeConfig = async (configId) => {
+    const res = await UserConfigurationsModel.destroy(uid, configId)
+    if (res.status === 200) {
+      setConfigurations((configurations) =>
+        configurations.filter(({ _id }) => _id !== configId)
+      )
+
+      if (checked) updateUserPreferences(configId)
+    }
+  }
+
+  const updateConfiguration = async (configId, configAttributes) => {
+    const { status } = await UserConfigurationsModel.update(
+      uid,
+      configId,
+      configAttributes
+    )
+
+    switch (true) {
+      case status === 200: {
+        loadAllConfigurations(uid)
+        break
+      }
+      case checked: {
+        updateUserPreferences(configId)
+        break
+      }
+      default:
+        break
+    }
+  }
+
+  const updateUserPreferences = async (configId) => {
+    const userAttributes = {
+      preferences: {
+        ...preferences,
+        config: preferences.config === configId ? '' : configId,
+      },
+    }
+
+    await UserModel.update(uid, userAttributes)
+
+    setPreferences(userAttributes.preferences)
+  }
+
+  return (
+    <Card style={{ margin: '3px', width: `${width}px` }}>
+      <CardHeader
+        title={owner}
+        subheader={updated}
+        avatar={<ConfigCardAvatar config={config} currentUser={user} />}
+        action={
+          <IconButton
+            onClick={() => {
+              if (ownsConfig) {
+                removeConfig(_id)
+              } else {
+                const configAttributes = {
+                  readers: readers.filter((reader) => reader !== uid),
+                }
+                updateConfiguration(_id, configAttributes)
+              }
+            }}
+          >
+            <Clear color={colors.gray} />
+          </IconButton>
+        }
+      />
+      <Divider />
+      <div className={classes.actionsDivider}>
+        <Typography variant="headline" component="h3" noWrap>
+          {name}
+        </Typography>
+        <Typography className={classes.textAndIcon} component="p">
+          {type}
+        </Typography>
+      </div>
+      <CardActions className={classes.actionsContainer}>
+        <FormControlLabel
+          control={
+            <Switch
+              labelStyle={classes.textAndIcon}
+              checked={checked}
+              onChange={() => updateUserPreferences(_id)}
+            />
+          }
+          label="Default"
+        />
+        <div>
+          {ownsConfig ? (
+            <IconButton
+              onClick={() => openNewWindow(routes.editConfiguration(_id))}
+              iconStyle={classes.textAndIcon}
+              tooltipPosition="top-center"
+              tooltip="Edit"
+            >
+              <Edit />
+            </IconButton>
+          ) : (
+            <IconButton
+              onClick={() => openNewWindow(routes.viewConfiguration(_id))}
```

Comment:
should these be links instead of buttons?

Reply:
Yeah they lead to different pages. This will change on edit config page refactors and after client side routing is implementing